File: lua/WorldClock.lua
--  Copyright 2012, John Wilson, Brighton Sussex UK. Licensed under the BSD License. See licence.txt
if (not WorldClock) then

WorldClock = {}
WorldClock.time = 0
WorldClock.observers = {}	
WorldClock.pause	= false

	-- Shared GameClock - for debouncing 
	
	-- Allow Watchers 
	function WorldClock.AddObserver(func)
		if (type(func)=="function") then
			table.insert(WorldClock.observers,func)	
		end
	end
	
	function WorldClock.Restart()
		WorldClock.Resume()
		WorldClock.Reset()
	end
	
	function WorldClock.Reset()
		WorldClock.time = 0
		WorldClock.InformObservers()
	end
	
	function WorldClock.Pause()
		WorldClock.pause = true 
	end
	
	function WorldClock.Resume()
		WorldClock.pause = false
	end
	
	-- Inform Watchers of 'major' changes in time
	function WorldClock.InformObservers()
		for k,f in pairs(WorldClock.observers) do
			if (f and type(f) == "function") then
				f(WorldClock.time)
			end
		end
	end
	
	function WorldClock.GetClock()
		return WorldClock.time 
	end

	function WorldClock.Update(dtime)
		if (not WorldClock.pause) then
			WorldClock.time = WorldClock.time + dtime
		end
	end

	Logger.print("Included WorldClock")

end

File: source/library/WorldClock.js
//Copyright 2012, John Wilson, Brighton Sussex UK. Licensed under the BSD License. See licence.txt

WorldClock 				=	{}	;
WorldClock.time 		=	0 ;
WorldClock.observers	=	new Array()	;
WorldClock.pause		=	false ;

WorldClock.AddObserver = function(func)
{
	WorldClock.observers.push(func)	;
}
	
WorldClock.Restart = function()
{
		WorldClock.Resume()
		WorldClock.Reset()
}
	
WorldClock.Reset = function()
{
	WorldClock.time = 0
	WorldClock.InformObservers()
	
}

WorldClock.Pause = function()
{
	WorldClock.pause = true 
	
}

WorldClock.Resume	=	function()
{
	WorldClock.pause = false
	
}


//Inform Watchers of 'major' changes in time
WorldClock.InformObservers = function()
{
		for (var oidx in WorldClock.observers) 
		{
			var f = 	WorldClock.observers[oidx] ;
			if (f) {
				f(WorldClock.time)	;
			}
		}
}
	
WorldClock.GetClock = function()
{
	return WorldClock.time ;
	
}

WorldClock.Update = function(dtime)
{
	if (!WorldClock.pause) {
		WorldClock.time = WorldClock.time + dtime
	}
}


